### PR TITLE
fix #617: a few asks from VM team during command review

### LINKED ISSUE #617: Omnibus of compute review asks
_Please feel free to break out any part of this issue as required._

* ~~For cases where we have [id,name,rg] such as `az vm boot-diagnostics enable`, we should have a pattern for making this obvious~~. This is a general issue: Open #656
* Rename `usage list` to `list-usages` to match xplat and prevent a single-item group
* We should break out `--expand instanceView` from `az vm show` **or** provide an enum in help (and tab completion).
  * Another option is to create a `beta-show-instanceview` to make it clear this command only exists until the API supports a better pattern, as breaking out all expands across all RPs is a bit much. 
* Allow for masked interactive prompting for `--admin-password` during `vm create`
  * Show if a password is required but not supplied
  * Show if an empty `--admin-password` is supplied, such as `az vm create --admin-password --image ubuntults`
* Access extension - we should call out the rights of users in the help text

### PR DESCRIPTION
**Pull request by yugangw-msft**

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py [hunk 1]
```diff
@@ -163,7 +163,8 @@ def _handle_auth_types(**kwargs):
         if args.ssh_dest_key_path:
             raise CLIError('SSH parameters cannot be used with password authentication type')
         elif not args.admin_password:
-            raise CLIError('Admin password is required with password authentication type')
+            import getpass
+            args.admin_password = getpass.getpass('Admin Password: ')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py]: i will open an issue that we should still error out when running under scripting mode.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/generated.py [hunk 2]
```diff
@@ -55,6 +55,8 @@
 cli_command('vm generalize', VirtualMachinesOperations.generalize, factory)
 cli_command('vm show', VirtualMachinesOperations.get, factory, simple_output_query="{Name:name, ResourceGroup:resourceGroup, Location:location, VmSize:hardwareProfile.vmSize, OsType: storageProfile.osDisk.osType}")
 cli_command('vm list-vm-resize-options', VirtualMachinesOperations.list_available_sizes, factory, simple_output_query="[*].{Name: name, NumberOfCores: numberOfCores, MemoryInMb: memoryInMb, MaxDataDiskCount: maxDataDiskCount, ResourceDiskSizeInMb: resourceDiskSizeInMb, OsDiskSizeInMb: osDiskSizeInMb} | sort_by(@, &Name)")
+cli_command('vm get-instance-view', get_instance_view,
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/generated.py]: >get-instance-vie [](start = 16, length = 16)

To be consistent with other commands, shouldn't this be 'show-instance-view'?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py [hunk 3]
```diff
@@ -81,6 +81,16 @@
...
             """
 
+helps['vm access set-linux-user'] = """
+            type: command
+            long-summary: Note, the user will have an admin's priviledge.
+            """
+
+helps['vm access reset-windows-admin'] = """
+            type: command
+            long-summary: Note, this resets the admin's credentails. You can't add a new admin.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py]: >credentails. [](start = 56, length = 12)

Typo: credentials

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/generated.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/generated.py]: I suggest we leave it as it is till hear feedback. 
Computer team likes the current name, and VMSS is also using the same name.

**Comment by yugangw-msft** [comment]
@burtbiel, could you also take a look?

**Comment by derekbekoe** [comment]
:shipit:

**Pull request merged**